mux_client_impl/src/command.rs

```diff
@@ -0,0 +1,251 @@
+use super::RemoteChild;
+use super::Result;
+use super::{as_raw_fd, ChildStderr, ChildStdin, ChildStdout, Stdio};
+
+use std::path::PathBuf;
+use std::process;
+
+use openssh_mux_client::connection::{Connection, EstablishedSession, Session};
+
+/// A remote process builder, providing fine-grained control over how a new remote process should
+/// be spawned.
+///
+/// A default configuration can be generated using [`Session::command(program)`](Session::command),
+/// where `program` gives a path to the program to be executed. Additional builder methods allow
+/// the configuration to be changed (for example, by adding arguments) prior to spawning.  The
+/// interface is almost identical to that of [`std::process::Command`].
+///
+/// `Command` can be reused to spawn multiple remote processes. The builder methods change the
+/// command without needing to immediately spawn the process. Similarly, you can call builder
+/// methods after spawning a process and then spawn a new process with the modified settings.
+///
+/// # Environment variables and current working directory.
+///
+/// You'll notice that unlike its `std` counterpart, `Command` does not have any methods for
+/// setting environment variables or the current working directory for the remote command. This is
+/// because the SSH protocol does not support this (at least not in its standard configuration).
+/// For more details on this, see the `ENVIRONMENT` section of [`ssh(1)`]. To work around this,
+/// give [`env(1)`] a try. If the remote shell supports it, you can also prefix your command with
+/// `["cd", "dir", "&&"]` to run the rest of the command in some directory `dir`.
+///
+/// # Exit status
+///
+/// The `ssh` command generally forwards the exit status of the remote process. The exception is if
+/// a protocol-level error occured, in which case it will return with exit status 255. Since the
+/// remote process _could_ also return with exit status 255, we have no reliable way to distinguish
+/// between remote errors and errors from `ssh`, but this library _assumes_ that 255 means the
+/// error came from `ssh`, and acts accordingly.
+///
+///   [`ssh(1)`]: https://linux.die.net/man/1/ssh
+///   [`env(1)`]: https://linux.die.net/man/1/env
+#[derive(Debug)]
+pub struct Command<'s> {
```

Comment:
Same thing here as for `RemoteChild`. I'd like to keep just _one_ type so that we keep the documentation and methods all the same, and then just delegate to a generic inner type.